### ISSUE TITLE
Parameters setting

### ISSUE THREAD

**Issue by JesseYANK** [jaredleekatzman/DeepSurv#3]
Any suggestions for parameters setting. The performance of default setting is very poor

**Comment by jaredleekatzman** [comment]
Sorry for the confusion, the default parameters are place holders and are
not meant to produce a working network.

I am not able to suggest parameters as optimal parameters are specific to
whichever dataset you are using.

I suggest running a parameter search algorithm. For example use [optunity](
http://optunity.readthedocs.io/en/latest/) for a random hyperparameter
search.

**Comment by LeZhengThu** [comment]
@jaredleekatzman  Hi, I tried your code and the network.predict_risk(X) gave output nan. Can you help explain why this happens? Thanks.

**Comment by jaredleekatzman** [comment]
@LeZhengThu could you provide more context? Did you train the network? Is it performing well on a validation set?

**Comment by JesseYANK** [comment]
@LeZhengThu 

You should check your input data. If the some variables are extremely large when compared to other variables, you will get nan.

You can normalise the input data, which should work for your case.

**Comment by LeZhengThu** [comment]
@JesseYANK @jaredleekatzman  Thank you for the reply. I'll try @JesseYANK method to normalise the data and get back to you guys.

**Comment by abracadabrac** [comment]
I had the same issue as LeZhengThu. I tried JessegThu method consisted is normalizing my data :
x = ( x - mean(x) ) / 1000 
It seems to works now.

**Comment by dareneiri** [comment]
@jaredleekatzman Can you please provide example of how you implemented Optunity for random hyperparameter search that is specific to deepsurv?

**Comment by jaredleekatzman** [comment]
Hi, I am currently working on releasing the Optunity script so that people can run their own hyper-parameter search. I hope to have it done soon.

**Comment by jaredleekatzman** [comment]
@dareneiri Sorry for the delay, but I've uploaded the script I used to run a random hyper-parameter search under the directory `hyperparam_search/`.
If you still need to use it, let me know how it goes.

**Comment by hanxiaozhen2017** [comment]
@jaredleekatzman  I will try the random hyper-parameter search and do we need to normalize our data before this step? Many thanks!
